src/coreclr/jit/emitarm64.cpp

```diff
@@ -21738,6 +21922,10 @@ void emitter::emitIns_Call(EmitCallType          callType,
             assert(isValidUimm5From1(imm));
             return (32 - imm);
 
+        case INS_OPTS_SCALABLE_D:
```

Comment:
Do the existing uses of `insGetImmDiff` assume only B, H, S? If so, we should either put asserts before calling `insGetImmDiff` ensuring that we don't accidently pass D, **or** just have two different functions. This is a similar suggestion to `insEncodeSveElemsize_tszh_22_tszl_20_to_19` that I had.